src/cron/OrderInvalidation.ts

```diff
@@ -0,0 +1,63 @@
+import cron from "node-cron";
+import { ACCEPTED_ERROR_CODES } from "@hypercerts-org/marketplace-sdk";
+import { SupabaseDataService } from "../services/SupabaseDataService.js";
+import _ from "lodash";
+import { kyselyData } from "../client/kysely.js";
+import { sql } from "kysely";
+
+export default class OrderInvalidationCronjob {
+  private static instance: OrderInvalidationCronjob;
+  private dataService: SupabaseDataService;
+
+  private constructor() {
+    this.dataService = new SupabaseDataService();
+    this.setupCronJob();
+  }
+
+  private setupCronJob() {
+    // Run every 30 seconds
+    cron.schedule("*/30 * * * * *", async () => {
+      try {
+        await this.invalidateOrders();
+      } catch (error) {
+        console.error("Error in order invalidation cron job:", error);
+      }
+    });
+  }
+
+  public static start(): void {
+    if (!OrderInvalidationCronjob.instance) {
+      OrderInvalidationCronjob.instance = new OrderInvalidationCronjob();
+    }
+  }
+
+  private async invalidateOrders() {
+    const result = await kyselyData
+      .selectFrom("marketplace_orders")
+      .select(["itemIds", "invalidated", "validator_codes", "chainId"])
+      .where((eb) =>
+        eb.or([
+          eb("invalidated", "=", false),
+          // Include all orders that have been invalidated but only have validator codes that are in the accepted validator codes
```

Comment:
This sounds like a one of this? Do we want to run that every time?

Reply:
The example would be an order that has been invalidated as "Haven't started yet". We're currently not marking those as invalidated, but I think in the future we should. We didn't have many of those before, as we were defaulting to the current time as a starting time, and not many people were using the UI to change the sale starting time. With the default now set to 15 minutes from now, and more people starting to use the marketplace I think it makes sense to actually mark them as invalid and periodically check if they've started already.

I'll update the comment as I agree that explanation shouldn't be there and is confusing.

Reply:
Removed the comment

Reply:
Wouldn't it be nice to have a distilled version of your explanation above as the comment? The next person reading this code (or you in 6 months) will appreciate it, I'm sure.

Reply:
To add to this, we should mark them as invalid, no? We're not the only one creating orders so should inform the business logic of the API based on the implementation in our app.

Would that need to be a separate issue or part of this?